rust/ruby-prism/src/lib.rs

```diff
@@ -198,6 +198,18 @@ impl<'pr> ParseResult<'pr> {
         }
     }
 
+    /// Returns an optional location of the __END__ marker and the rest of the content of the file.
+    #[must_use]
+    pub fn data_loc(&self) -> Option<Location<'_>> {
+        let root = self.source.as_ptr();
+        let location = unsafe { &(*self.parser.as_ptr()).data_loc };
+        if location.start >= root {
```

Comment:
In other optional locations we're checking `start.is_null()` instead, can you do that here?

Reply:
Thank you, I changed the implementation to use `is_null()` .
https://github.com/ruby/prism/pull/2854/commits/be34f68fa502dfdbb2ba43454d343a2b8cd7c0f7